src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -694,7 +694,18 @@ private int TryPopCore(int count, out Node poppedHead)
                     spin.SpinOnce();
                 }
 
-                backoff = spin.NextSpinWillYield ? r.Next(1, BACKOFF_MAX_YIELDS) : backoff * 2;
+                if (spin.NextSpinWillYield)
+                {
+                    if (r == null)
+                    {
+                        r = new Random();
```

Comment:
Yes.  When this code was written, it was written assuming that Random's parameterless ctor used Environment.TickCount as the seed.  That has since been changed in coreclr.